system/Cache/Handlers/RedisHandler.php

```diff
@@ -228,25 +228,33 @@ public function delete(string $key)
 	 *
 	 * @param string $pattern Cache items glob-style pattern
 	 *
-	 * @return boolean
+	 * @return integer $deleted The number of deleted items
 	 */
 	public function deleteMatching(string $pattern)
 	{
-		$success  = true;
+		$deleted  = 0;
 		$iterator = null;
 
-		while ($keys = $this->redis->scan($iterator, $pattern))
+		do
 		{
-			foreach ($keys as $key)
+			// Scan for some keys
+			$keys = $this->redis->scan($iterator, $pattern);
+
+			// Redis may return empty results, so protect against that
+			if ($keys !== false)
 			{
-				if ($this->redis->del($key) !== 1)
+				foreach ($keys as $key)
 				{
-					$success = false;
+					if ($this->redis->del($key) === 1)
```

Comment:
`del` seems to accept an array of keys as well as its arguments. Why did you not use that?

Reply:
Because I didn't see that 😅 Thanks for the heads up!
I also optimized the FileHandler implementation by adding a `GLOB_NOSORT` flag to the `glob()` function: no need to sort the files before deleting them... It should be faster now!